webpack/assets/javascripts/react_app/components/common/forms/helpers.js

```diff
@@ -0,0 +1,14 @@
+import { noop } from '../../../common/helpers';
+import { deprecate } from '../../../common/DeprecationService';
+
+const getEventValue = ({ target: { type, checked, value } }) =>
+  type === 'checkbox' ? checked : value;
+
+export const makeOnChangeHanler = (onChange, onValueChange) => {
+  if (onChange !== noop) deprecate('onChange', 'onValueChange', '1.24');
```

Comment:
why can't we use the regular `onChange` ?

Reply:
I am for supporting both, but @sharvit were suggesting it would mean to support two APIs of the components, which could get overhelming. But I believe the deprecation is not necessary. At least not yet.